test/e2e/framework/networking_utils.go

```diff
@@ -255,10 +256,15 @@ func (config *NetworkingTestConfig) GetEndpointsFromContainer(protocol, containe
 					eps.Insert(trimmed)
 				}
 			}
-			return eps, nil
+			// Return immediately when we successfully fetch endpoints
+			if len(eps) > 0 {
+				return eps, nil
+			}
+			// TODO: get rid of this delay #36281
+			time.Sleep(hitEndpointRetryDelay)
 		}
 	}
-	return nil, fmt.Errorf("Failed to get endpoints:\nTries %d\nCommand %v\n", minTries, cmd)
+	return nil, fmt.Errorf("error getting endpoints:\nTries %d\nCommand %v\n", minTries, cmd)
```

Comment:
Seems we are not really using `minTries` in this fuction? Could we remove `minTries` and rename `maxTries` to `tries`?